File: ReactAndroid/src/main/java/com/facebook/react/packagerconnection/RequestOnlyHandler.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.packagerconnection;

import androidx.annotation.Nullable;
import com.facebook.common.logging.FLog;

public abstract class RequestOnlyHandler implements RequestHandler {
  private static final String TAG = JSPackagerClient.class.getSimpleName();

  public abstract void onRequest(@Nullable Object params, Responder responder);

  public final void onNotification(@Nullable Object params) {
    FLog.e(TAG, "Notification is not supported");
  }
}

File: packages/react-native/temp/package/ReactAndroid/src/main/java/com/facebook/react/packagerconnection/RequestOnlyHandler.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.packagerconnection

import com.facebook.common.logging.FLog

public abstract class RequestOnlyHandler : RequestHandler {

  override abstract fun onRequest(params: Any?, responder: Responder)

  override final fun onNotification(params: Any?) {
    FLog.e(JSPackagerClient::class.java.simpleName, "Notification is not supported")
  }
}
